ci/docker/ubuntu-20.04-cpp.dockerfile

```diff
@@ -129,6 +130,7 @@ ENV ARROW_BUILD_TESTS=ON \
     ARROW_WITH_ZSTD=ON \
     ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-${llvm}/bin/llvm-symbolizer \
     AWSSDK_SOURCE=BUNDLED \
+    google_cloud_cpp_storage_SOURCE=BUNDLED \
```

Comment:
Was `GCS_SOURCE` too ambiguous? This is quite a long parameter name...

Reply:
We use CMake's package name for `XXX_SOURCE`.
If we don't follow the convention, we can choose `GCS_SOURCE`.

Reply:
Let me know how to proceed, I can do either.

Reply:
If it's the CMake name then fair enough.